generics/services/users.js

```diff
@@ -6,350 +6,370 @@
  */
 
 //dependencies
-const request = require('request');
+const request = require("request");
 const userServiceUrl = process.env.USER_SERVICE_URL;
 
-const profile = function ( token,userId = "" ) {
-    return new Promise(async (resolve, reject) => {
-        try {
+const profile = function (token, userId = "") {
+  return new Promise(async (resolve, reject) => {
+    try {
+      let url = userServiceUrl + messageConstants.endpoints.USER_READ_V5;
 
-            let url = userServiceUrl + messageConstants.endpoints.USER_READ_V5;
-            
-            if( userId !== "" ) {
-                url = url + "/" + userId + "?"  + "fields=organisations,roles,locations,declarations,externalIds"
-            }
-            
-            const options = {
-                headers : {
-                    "content-type": "application/json",
-                    "x-authenticated-user-token" : token
-                }
-            };
-            
-            request.get(url,options,userReadCallback);
-            let result = {
-                success : true
-            };
-            function userReadCallback(err, data) {
-                
-                if (err) {
-                    result.success = false;
-                } else {
-                    let response = JSON.parse(data.body);
-                    if( response.responseCode === httpStatusCode['ok'].code ) {
-                        result["data"] = response.result;
-                    } else {
-                        result.success = false;
-                    }
-
-                }
-                
-                return resolve(result);
-            }
-            setTimeout(function () {
-                return resolve (result = {
-                    success : false
-                });
-             }, messageConstants.common.SERVER_TIME_OUT);
-
-        } catch (error) {
-            return reject(error);
+      if (userId !== "") {
+        url =
+          url +
+          "/" +
+          userId +
+          "?" +
+          "fields=organisations,roles,locations,declarations,externalIds";
+      }
+
+      const options = {
+        headers: {
+          "content-type": "application/json",
+          "x-authenticated-user-token": token,
+        },
+      };
+
+      request.get(url, options, userReadCallback);
+      let result = {
+        success: true,
+      };
+      function userReadCallback(err, data) {
+        if (err) {
+          result.success = false;
+        } else {
+          let response = JSON.parse(data.body);
+          if (response.responseCode === httpStatusCode["ok"].code) {
+            result["data"] = response.result;
+            result.data = _.omit(response.result, [
+              "response.email",
```

Comment:
@ankitpws check if there is a flag to pass so that we will not get only any user pvt data